common/src/main/java/com/wynntils/features/EmeraldPouchHotkeyFeature.java

```diff
@@ -113,63 +118,63 @@ public class EmeraldPouchHotkeyFeature extends Feature {
-                                    largest = entry;
+                        if (!hasFilled) {
+                            EmeraldPouch largest = null;
+                            for (EmeraldPouch ep : emeraldPouches) {
+                                if (largest == null || ep.getCapacity() > largest.getCapacity()) {
+                                    largest = ep;
                                 }
                             }
-                            slotNumber = largest.getKey();
-                            slotStack = largest.getValue().a;
+                            slotNumber = largest.getSlotNumber();
+                            slotStack = largest.getStack();
                         }
 
                         // Now, we know we have 1 used and 1+ empty pouches - open the used one we saved from before
-                        if (slotNumber == -1) {
-                            slotNumber = usedPouch;
-                        }
-                        if (slotStack == null) {
-                            slotStack = pouchStack;
-                        }
                         if (slotNumber < 9) {
-                            // sendPacket uses raw slot numbers, we need to remap the hotbar
-                            slotNumber += 36;
+                            slotNumber += 36; // Raw slot numbers, remap if in hotbar
                         }
-                        changedSlots.putIfAbsent(slotNumber, slotStack);
-                        McUtils.player()
-                                .connection
-                                .send(new ServerboundContainerClickPacket(
-                                        player.inventoryMenu.containerId,
-                                        player.inventoryMenu.getStateId(),
-                                        slotNumber,
-                                        1,
-                                        ClickType.PICKUP,
-                                        ItemStack.EMPTY,
-                                        changedSlots));
+
+                        dispatchRightClick(slotNumber, slotStack);
                     }
                 }
             });
 
-    private int getPouchUsage(ItemStack stack) { // TODO: move to EmeraldPouchManager or equivalent when created
+    private static void dispatchRightClick(int slotNumber, ItemStack stack) {
+        Int2ObjectMap<ItemStack> changedSlots = new Int2ObjectOpenHashMap<>();
+        changedSlots.putIfAbsent(slotNumber, stack);
+        McUtils.player()
+                .connection
+                .send(new ServerboundContainerClickPacket(
+                        McUtils.player().inventoryMenu.containerId,
+                        McUtils.player().inventoryMenu.getStateId(),
+                        slotNumber,
+                        1,
+                        ClickType.PICKUP,
+                        ItemStack.EMPTY,
+                        changedSlots));
```

Comment:
`putIfAbsent` should be a `put` call. Also, pretty sure this can be further inlined, but its up to you
```java
        McUtils.player()
                .connection
                .send(new ServerboundContainerClickPacket(
                        McUtils.player().inventoryMenu.containerId,
                        McUtils.player().inventoryMenu.getStateId(),
                        slotNumber,
                        1,
                        ClickType.PICKUP,
                        ItemStack.EMPTY,
                        new Int2ObjectOpenHashMap<>() {{ put(slotNumber, stack) }}));
```

Reply:
Done - I originally used putIfAbsent because one of the `put` methods were deprecated, but I found the one that isn't